pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java

```diff
@@ -1151,28 +1151,29 @@ public void testDeleteTopic() throws Exception {
         BrokerService brokerService = spy(pulsar.getBrokerService());
         doReturn(brokerService).when(pulsar).getBrokerService();
         persistentTopics.createNonPartitionedTopic(testTenant, testNamespace, topicName, false, null);
+
         CompletableFuture<Void> deleteTopicFuture = new CompletableFuture<>();
         deleteTopicFuture.completeExceptionally(new MetadataStoreException.NotFoundException());
         doReturn(deleteTopicFuture).when(brokerService).deleteTopic(anyString(), anyBoolean(), anyBoolean());
-        persistentTopics.deleteTopic(testTenant, testNamespace, topicName, true, true, true);
-        //
+        AsyncResponse response = mock(AsyncResponse.class);
+        ArgumentCaptor<Response> responseCaptor = ArgumentCaptor.forClass(Response.class);
+        persistentTopics.deleteTopic(response, testTenant, testNamespace, topicName, true, true, true);
+        verify(response, timeout(5000).times(1)).resume(responseCaptor.capture());
+
+        ArgumentCaptor<RestException> errorCaptor = ArgumentCaptor.forClass(RestException.class);
         CompletableFuture<Void> deleteTopicFuture2 = new CompletableFuture<>();
         deleteTopicFuture2.completeExceptionally(new MetadataStoreException("test exception"));
         doReturn(deleteTopicFuture2).when(brokerService).deleteTopic(anyString(), anyBoolean(), anyBoolean());
-        try {
-            persistentTopics.deleteTopic(testTenant, testNamespace, topicName, true, true, true);
-        } catch (Exception e) {
-            Assert.assertTrue(e instanceof RestException);
-        }
-        //
+        response = mock(AsyncResponse.class);
+        persistentTopics.deleteTopic(response, testTenant, testNamespace, topicName, true, true, true);
+        verify(response, timeout(5000).times(1)).resume(errorCaptor.capture());
+
```

Comment:
Please check the `errorCaptor.getValue()`.

Reply:
`errorCaptor` already checks that instance type.